vars/containerPushToRegistry.groovy

```diff
@@ -0,0 +1,129 @@
+     * Simple custom Dockerfile could look as follows: <br>
+     * ```
+     * FROM fedora:29
+     * RUN dnf install -y skopeo
+     * ```
+     */
+    'skopeoImage',
+    /** Defines the name (incl. tag) of the source image to be pushed to a new image defined in `dockerImage`.<br>
+     * This is helpful for moving images from one location to another.
+     */
+    'sourceImage',
+    /** Defines a registry url from where the image should optionally be pulled from, incl. the protocol like `https://my.registry.com`*/
+    'sourceRegistryUrl',
+    /** Defines if the image should be tagged as `latest`*/
+    'tagLatest'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * This step allows you to push a Docker image into a dedicated Container registry.
+ *
+ * By default an image available via the local Docker daemon will be pushed.
+ *
+ * In case you want to pull an existing image from a remote container registry, a source image and source registry needs to be specified.<br />
+ * This makes it possible to move an image from one registry to another.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .addIfEmpty('sourceImage', script.commonPipelineEnvironment.getValue('containerImage'))
+            .addIfEmpty('sourceRegistryUrl', script.commonPipelineEnvironment.getValue('containerRegistryUrl'))
+            .withMandatoryProperty('dockerCredentialsId')
+            .withMandatoryProperty('dockerRegistryUrl')
+            .use()
+
+        DockerUtils dockerUtils = new DockerUtils(script)
+
+        if (config.sourceRegistryUrl) {
+            config.sourceRegistry = dockerUtils.getRegistryFromUrl(config.sourceRegistryUrl)
+        }
+
+        // telemetry reporting
+        new Utils().pushToSWA([
+            step: STEP_NAME
+        ], config)
+
+        if (!config.dockerImage)
+            config.dockerImage = config.sourceImage
+
+        if (dockerUtils.withDockerDaemon()) {
+
+            //Prevent NullPointerException in case no dockerImage nor dockerBuildImage is provided
+            if (!config.dockerImage && !config.dockerBuildImage) {
+                error "[${STEP_NAME}] Please provide a dockerImage (either in your config.yml or via step parameter)."
+            }
+            config.dockerBuildImage = config.dockerBuildImage?:docker.image(config.dockerImage)
```

Comment:
assumption is that config is a local map, as in many other steps as well.
Since it is anyway just a String it should not harm, or does it?
Do you see any concrete issue?